test/test-tcp-exportimport.c

```diff
@@ -0,0 +1,237 @@
+/* Copyright Joyent, Inc. and other Node contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "runner.h"
+#include "task.h"
+
+#include <stdio.h>
+#include <string.h>
+
+typedef struct {
+  uv_loop_t* loop;
+  uv_thread_t thread;
+  uv_async_t* recv_channel;
+  uv_async_t* send_channel;
+  uv_tcp_t server;
+  uv_tcp_t conn;
+  int connection_accepted;
+  int close_cb_called;
+} worker_t;
+
+static uv_async_t send_channel;
+static uv_async_t recv_channel;
+static worker_t parent;
+static worker_t child;
+
+static uv_stream_info_t dup_stream;
+
+typedef struct {
+  uv_connect_t conn_req;
+  uv_tcp_t conn;
+} tcp_conn;
+
+#define CONN_COUNT 100
+static tcp_conn conns[CONN_COUNT];
+
+static void close_cb(uv_handle_t* handle) {
+  worker_t* worker = (worker_t*)handle->data;
+  ASSERT(worker != NULL);
+  worker->close_cb_called++;
+}
+
+
+static void on_connection(uv_stream_t* server, int status) {
+  int r;
+  worker_t* worker = container_of(server, worker_t, server);
+
+  if (!worker->connection_accepted) {
+    /*
+     * Accept the connection and close it.
+     */
+    ASSERT(status == 0);
```

Comment:
```suggestion
    ASSERT_EQ(status, 0);
```
(I'll stop pointing these out from here on but please update them all.)